StRoot/StFcsWaveformFitMaker/StFcsPulseAna.cxx

```diff
@@ -640,25 +645,26 @@ void StFcsPulseAna::SetFitPars(TF1* func)
   para[1] = Baseline();
   func->SetParName(0,"NPeaks");
   func->SetParName(1,"Ped");
-  for( Int_t i=0; i<npeaks; ++i ){
+  for( int i=0; i<npeaks; ++i ){
     int j = 1+i*3+1;
     char name[10];
-    sprintf(name,"P%d_A",i);
+    sprintf(name,"P%d_A",start+i);
     func->SetParName(j,name);
-    para[j++] = GetPeak(i).mPeakY;
-    sprintf(name,"P%d_M",i);
+    para[j++] = GetPeak(start+i).mPeakY;
+    sprintf(name,"P%d_M",start+i);
     func->SetParName(j,name);
-    para[j++] = GetPeak(i).mPeakX;
-    sprintf(name,"P%d_S",i);
+    para[j++] = GetPeak(start+i).mPeakX;
+    sprintf(name,"P%d_S",start+i);
```

Comment:
This change introduced compiler warnings. Please address

```
2023-01-10T23:20:47.9029308Z #10 1302.4 .sl79_gcc11/OBJ/StRoot/StFcsWaveformFitMaker/StFcsPulseAna.cxx: In member function 'void StFcsPulseAna::SetFitPars(TF1*&, int, int)':
2023-01-10T23:20:47.9030257Z #10 1302.4 .sl79_gcc11/OBJ/StRoot/StFcsWaveformFitMaker/StFcsPulseAna.cxx:651:20: warning: '%d' directive writing between 1 and 10 bytes into a region of size 9 [-Wformat-overflow=]
2023-01-10T23:20:47.9030759Z #10 1302.4   651 |     sprintf(name,"P%d_A",start+i);
2023-01-10T23:20:47.9031024Z #10 1302.4       |                    ^~
2023-01-10T23:20:47.9031443Z #10 1302.4 .sl79_gcc11/OBJ/StRoot/StFcsWaveformFitMaker/StFcsPulseAna.cxx:651:18: note: directive argument in the range [0, 2147483647]
2023-01-10T23:20:47.9031866Z #10 1302.4   651 |     sprintf(name,"P%d_A",start+i);
2023-01-10T23:20:47.9032149Z #10 1302.4       |                  ^~~~~~~
2023-01-10T23:20:47.9032765Z #10 1302.4 .sl79_gcc11/OBJ/StRoot/StFcsWaveformFitMaker/StFcsPulseAna.cxx:651:12: note: 'sprintf' output between 5 and 14 bytes into a destination of size 10
2023-01-10T23:20:47.9033215Z #10 1302.4   651 |     sprintf(name,"P%d_A",start+i);
2023-01-10T23:20:47.9033503Z #10 1302.4       |     ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
2023-01-10T23:20:47.9034178Z #10 1302.4 .sl79_gcc11/OBJ/StRoot/StFcsWaveformFitMaker/StFcsPulseAna.cxx:654:20: warning: '%d' directive writing between 1 and 10 bytes into a region of size 9 [-Wformat-overflow=]
2023-01-10T23:20:47.9034662Z #10 1302.4   654 |     sprintf(name,"P%d_M",start+i);
2023-01-10T23:20:47.9034935Z #10 1302.4       |                    ^~
2023-01-10T23:20:47.9035345Z #10 1302.4 .sl79_gcc11/OBJ/StRoot/StFcsWaveformFitMaker/StFcsPulseAna.cxx:654:18: note: directive argument in the range [0, 2147483647]
2023-01-10T23:20:47.9035751Z #10 1302.4   654 |     sprintf(name,"P%d_M",start+i);
2023-01-10T23:20:47.9036030Z #10 1302.4       |                  ^~~~~~~
2023-01-10T23:20:47.9036656Z #10 1302.4 .sl79_gcc11/OBJ/StRoot/StFcsWaveformFitMaker/StFcsPulseAna.cxx:654:12: note: 'sprintf' output between 5 and 14 bytes into a destination of size 10
2023-01-10T23:20:47.9037090Z #10 1302.4   654 |     sprintf(name,"P%d_M",start+i);
2023-01-10T23:20:47.9037373Z #10 1302.4       |     ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
2023-01-10T23:20:47.9038379Z #10 1302.4 .sl79_gcc11/OBJ/StRoot/StFcsWaveformFitMaker/StFcsPulseAna.cxx:657:20: warning: '%d' directive writing between 1 and 10 bytes into a region of size 9 [-Wformat-overflow=]
2023-01-10T23:20:47.9038879Z #10 1302.4   657 |     sprintf(name,"P%d_S",start+i);
2023-01-10T23:20:47.9039140Z #10 1302.4       |                    ^~
2023-01-10T23:20:47.9039549Z #10 1302.4 .sl79_gcc11/OBJ/StRoot/StFcsWaveformFitMaker/StFcsPulseAna.cxx:657:18: note: directive argument in the range [0, 2147483647]
2023-01-10T23:20:47.9039968Z #10 1302.4   657 |     sprintf(name,"P%d_S",start+i);
2023-01-10T23:20:47.9040230Z #10 1302.4       |                  ^~~~~~~
2023-01-10T23:20:47.9040856Z #10 1302.4 .sl79_gcc11/OBJ/StRoot/StFcsWaveformFitMaker/StFcsPulseAna.cxx:657:12: note: 'sprintf' output between 5 and 14 bytes into a destination of size 10
2023-01-10T23:20:47.9041304Z #10 1302.4   657 |     sprintf(name,"P%d_S",start+i);
2023-01-10T23:20:47.9041591Z #10 1302.4       |     ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
```

Reply:
That's odd because I don't remember seeing this warning. I increased size of the char array and I don't see such warnings

Reply:
These are from GCC 11. You can see the build log if you go to the Checks tab
I see that these warnings are gone now. Thanks